api/controllers/FeedbackController.ts

```diff
@@ -0,0 +1,42 @@
+import { Body, ForbiddenError, Get, JsonController, Param, Patch, Post, UseBefore } from 'routing-controllers';
+import { Inject } from 'typedi';
+import { AuthenticatedUser } from '../decorators/AuthenticatedUser';
+import { UserModel } from '../../models/UserModel';
+import PermissionsService from '../../services/PermissionsService';
+import FeedbackService from '../../services/FeedbackService';
+import { GetFeedbackResponse, SubmitFeedbackResponse, UpdateFeedbackStatusResponse, Uuid } from '../../types';
+import { UserAuthentication } from '../middleware/UserAuthentication';
+import {
+  SubmitFeedbackRequest,
+  UpdateFeedbackStatusRequest,
+} from '../validators/FeedbackControllerRequests';
+
+@UseBefore(UserAuthentication)
+@JsonController('/feedback')
+export class FeedbackController {
+  @Inject()
+  private feedbackService: FeedbackService;
+
+  @Get()
+  async getFeedback(@AuthenticatedUser() user: UserModel): Promise<GetFeedbackResponse> {
+    const canSeeAllFeedback = PermissionsService.canAcknowledgeFeedback(user);
+    const feedback = await this.feedbackService.getFeedback(canSeeAllFeedback, user);
+    return { error: null, feedback };
+  }
+
+  @Post()
+  async submitFeedback(@Body() submitFeedbackRequest: SubmitFeedbackRequest,
+    @AuthenticatedUser() user: UserModel): Promise<SubmitFeedbackResponse> {
+    const feedback = await this.feedbackService.submitFeedback(user, submitFeedbackRequest.feedback);
```

Comment:
Do we want to restrict leaving feedback to users that're email-verified or something?